### PR TITLE
MC-21773: [MFTF] Failing Test: EndToEndB2CGuestUserMysqlTest

### PR DESCRIPTION
**Pull request by abothu**
"This E2E test is not part of Inventory Module but it is part of CE,EE,B2B module and uses the extension to test Inventory Module as well. We recently made changes to original test "EndToEndB2CGuestUserTest" as part of Elastic Search support. But at the same time we clone original test EndToEndB2CGuestUserTest and created a new one EndToEndB2CGuestUserMysqlTest to support the default Mysql Search Engine as well. While making changes we forgot to make changes of adding EndToEndB2CGuestUserMysqlTest on MSI side as well which causes our nightly jobs to fail as it runs on MSI extensions. I made necessary changes to fix this test"

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Catalog/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml [hunk 1]
```diff
@@ -17,6 +17,198 @@
             <description value="User browses catalog, searches for product, adds product to cart, adds product to wishlist, compares products, uses coupon code and checks out."/>
             <severity value="CRITICAL"/>
             <testCaseId value="MAGETWO-87435"/>
+            <group value="SearchEngineElasticsearch"/>
```
Comment (sidolov) [inline app/code/Magento/Catalog/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml:20]: Please, revert changes, files in the Catalog module are part of Magento core, not MSI project

File: app/code/Magento/CatalogSearch/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml [hunk 2]
```diff
@@ -65,6 +65,72 @@
         <grabAttributeFrom selector="{{StorefrontCategoryProductSection.ProductImageByName($$createSimpleProduct2.name$$)}}" userInput="src" stepKey="searchGrabSimpleProduct2ImageSrc" after="searchAssertFilterCategorySimpleProduct2"/>
         <assertNotRegExp expected="'/placeholder\/small_image\.jpg/'" actual="$searchGrabSimpleProduct2ImageSrc" stepKey="searchAssertSimpleProduct2ImageNotDefault" after="searchGrabSimpleProduct2ImageSrc"/>
 
+        <!-- Quick Search with non-existent product name -->
```
Comment (sidolov) [inline app/code/Magento/CatalogSearch/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml]: Please, revert changes, files in the CatalogSearch module are part of Magento core, not MSI project

File: app/code/Magento/Checkout/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml [hunk 3]
```diff
@@ -184,6 +184,194 @@
             <argument name="productVar" value="$$createSimpleProduct2$$"/>
         </actionGroup>
 
+        <comment userInput="Place order with check money order payment" stepKey="commentPlaceOrderWithCheckMoneyOrderPayment" after="guestCheckoutCheckSimpleProduct2InCartItems" />
```
Comment (sidolov) [inline app/code/Magento/Checkout/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml]: Please, revert changes, files in the Checkout module are part of Magento core, not MSI project

File: app/code/Magento/ConfigurableProduct/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml [hunk 4]
```diff
@@ -9,6 +9,215 @@
 <tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
     <test name="EndToEndB2CGuestUserTest">
+    <before>
```
Comment (sidolov) [inline app/code/Magento/ConfigurableProduct/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml:12]: Please, revert changes, files in the ConfigurableProduct module are part of Magento core, not MSI project

File: app/code/Magento/SalesRule/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml [hunk 5]
```diff
@@ -38,6 +38,45 @@
             <argument name="total" value="447.00"/>
         </actionGroup>
 
+        <actionGroup ref="StorefrontCancelCouponActionGroup" stepKey="couponCancelCoupon" after="couponCheckCartWithDiscount"/>
```
Comment (sidolov) [inline app/code/Magento/SalesRule/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml]: Please, revert changes, files in the SalesRule module are part of Magento core, not MSI project

File: dev/tests/acceptance/tests/functional/Magento/FunctionalTest/ConfigurableProductCatalogSearch/Test/EndToEndB2CGuestUserTest.xml [hunk 6]
```diff
@@ -27,4 +27,23 @@
         <grabAttributeFrom selector="{{StorefrontProductInfoMainSection.productImage}}" userInput="src" stepKey="searchGrabConfigProductPageImageSrc" after="searchAssertConfigProductPage"/>
         <assertNotRegExp expected="'/placeholder\/image\.jpg/'" actual="$searchGrabConfigProductPageImageSrc" stepKey="searchAssertConfigProductPageImageNotDefault" after="searchGrabConfigProductPageImageSrc"/>
     </test>
+    <test name="EndToEndB2CGuestUserMysqlTest">
```
Comment (sidolov) [inline dev/tests/acceptance/tests/functional/Magento/FunctionalTest/ConfigurableProductCatalogSearch/Test/EndToEndB2CGuestUserTest.xml]: Please, revert changes, files in the ConfigurableProductCatalogSearch module are part of Magento core, not MSI project

**Review by sidolov — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml (hunk 1 above)
Comment (abothu) [reply app/code/Magento/Catalog/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml:20]: Reverted changes

File: app/code/Magento/CatalogSearch/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml (hunk 2 above)
Comment (abothu) [reply app/code/Magento/CatalogSearch/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml]: Reverted changes

File: app/code/Magento/Checkout/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml (hunk 3 above)
Comment (abothu) [reply app/code/Magento/Checkout/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml]: Reverted changes

File: app/code/Magento/ConfigurableProduct/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml (hunk 4 above)
Comment (abothu) [reply app/code/Magento/ConfigurableProduct/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml:12]: Reverted changes

File: app/code/Magento/SalesRule/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml (hunk 5 above)
Comment (abothu) [reply app/code/Magento/SalesRule/Test/Mftf/Test/EndToEndB2CGuestUserTest.xml]: Reverted changes

File: dev/tests/acceptance/tests/functional/Magento/FunctionalTest/ConfigurableProductCatalogSearch/Test/EndToEndB2CGuestUserTest.xml (hunk 6 above)
Comment (abothu) [reply dev/tests/acceptance/tests/functional/Magento/FunctionalTest/ConfigurableProductCatalogSearch/Test/EndToEndB2CGuestUserTest.xml]: Reverted changes

**Comment by abothu** [comment]
Closing this PR and created a new one.

**Pull request closed without merging**